utils/src/main/kotlin/Utils.kt

```diff
@@ -567,3 +572,50 @@ fun Throwable.showStackTrace() {
     // https://discuss.kotlinlang.org/t/if-operator-in-function-expression/7227.
     if (printStackTrace) printStackTrace()
 }
+
+/**
+ * Return Closable which temporarily moves the given directories to a temporary directory on initialization. On close
+ * it moves them back to their original location overwriting any conflicting existing directory.
+ */
+fun stashDirectories(vararg directories: File) : Closeable {
+    val directorySet = HashSet(Arrays.asList(*directories))
+    return StashDirectory(directorySet)
+}
+
+private class StashDirectory(directories: Set<File>) : Closeable {
+    private val stashedDirectories = mutableMapOf<File, File>()
+
+    init {
+        directories.forEach() {
+            stashDir(it)
+        }
+    }
+
+    override fun close() {
+        HashMap(stashedDirectories).forEach() {
+            unstashDir(it.key)
+        }
+    }
+
+    private fun stashDir(originalDir: File) {
+        if (!originalDir.isDirectory) return
+
+        val tempDir = createTempDir("analyzer", ".tmp", originalDir.parentFile)
```

Comment:
We don't know whether this will still only be called from the analyzer, so we should use a more generic prefix. I'd prefer e.g. `originalDir.name`.